libqtile/core/manager.py

```diff
@@ -408,14 +408,78 @@ def update_net_desktops(self) -> None:
         self.core.update_net_desktops(self.groups, index)
 
     def update_client_list(self) -> None:
-        """Updates the client stack list
+        """Update X11 client list
 
         This is needed for third party tasklists and drag and drop of tabs in
         chrome
         """
         windows = [wid for wid, c in self.windows_map.items() if c.group]
         self.core.update_client_list(windows)
 
+    def update_client_order(self) -> None:
+        """Update X11 client order
+        This is needed for third party tasklists and drag and drop of tabs in
+        chrome
+        """
+        windows = [wid for wid, c in self.windows_map.items() if c.group]
+        stack = self.conn.query_tree()
+        # TODO: there appear to be some windows (starting VirtualBox creates one) that might not be properly unmanaged
+        windows = sorted([w for w in windows if w in stack], key=lambda wid: stack.index(wid))
+        self.core.update_client_order(windows)
```

Comment:
Could you move the X-specific logic into the core itself, keeping only general high-level logic in the manager?

Also with the stuff below.

Reply:
So... just call `query_tree` in the core and directly feed the result into the stacking order? Without anything in between? That would probably be much easier, actually.

Reply:
To keep the functionality as is (I'm not sure it would be a good idea to just feed the full stack back into X's stacking order), I just moved the three middle lines into the called core function with the latest change to the first commit. Let me know if you agree or disagree with it.